### PR TITLE
Add settings and menu items to toggle display of each section

### LINKED ISSUE #10: Allow to toggle which sections to show?
Currently there are four sections:
- Create Empty
- Starred
- Launch Notebook
- Launch Console

The Starred section is configurable (can be shown/hidden) based on early talks in the design phase. The question is, should user be able to toggle any of them in the settings?

### PR DESCRIPTION
**Pull request by andrewfulton9**
## Reference Issues or PRs
closes #10 
Adds settings for which sections to show. 

https://github.com/user-attachments/assets/b9acfa90-90d1-4d09-b256-a80a4e12ab77

## What does this implement/fix?

_Put a `x` in the boxes that apply_

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds a feature)
- [ ] Breaking change (fix or feature that would cause existing features not to work as expected)
- [ ] Documentation Update
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no API changes)
- [ ] Build related changes
- [ ] Other (please describe):

## Testing

- [x] Did you test the pull request locally?
- [x] Did you add new tests?

#### Text styling

- [ ] The content is written with [plain language](https://www.plainlanguage.gov/guidelines/) (where relevant).
- [ ] If there are headers, they use the proper header tags (with only one level-one header: `H1` or `#` in markdown).
- [ ] All links describe where they link to (for example, check the [Nebari website](https://nebari.dev/)).
- [ ] This content adheres to the Nebari style guides.

#### Non-text content

- [ ] All content is represented as text (for example, images need alt text, and videos need captions or descriptive transcripts).
- [ ] If there are emojis, there are not more than three in a row.
- [ ] Don't use [flashing GIFs or videos](https://www.w3.org/TR/UNDERSTANDING-WCAG20/seizure-does-not-violate.html).
- [ ] If the content were to be read as plain text, it still makes sense, and no information is missing.

### CODE REVIEW

File: schema/plugin.json [hunk 1]
```diff
@@ -56,10 +56,25 @@
         "type": "string"
       }
     },
+    "createEmptySection": {
+      "type": "boolean",
+      "title": "Show create empty section",
```
Comment (krassowski) [inline schema/plugin.json:61]: Would it be more readable if we added quotes and capitalization here?

```suggestion
      "title": "Show \"Create Empty\" section",
```

File: src/commands.ts [hunk 2]
```diff
@@ -76,6 +76,19 @@ export function addCommands(
       await settings.set('columnOrder', order);
     }
   });
+  app.commands.addCommand(CommandIDs.showCreateEmpty, {
+    isToggleable: true,
+    isToggled: () => {
+      return settings.composite
+        .createEmptySection as ISettingsLayout['createEmptySection'];
+    },
+    label: trans.__('Show Create Empty Section'),
```
Comment (krassowski) [inline src/commands.ts:85]: Same remark about adding quotes for labels.

File: src/commands.ts [hunk 3]
```diff
@@ -89,6 +102,32 @@ export function addCommands(
       await settings.set('starredSection', !starredSection);
     }
   });
+  app.commands.addCommand(CommandIDs.showNotebookLauncher, {
+    isToggleable: true,
+    isToggled: () => {
+      return settings.composite
+        .launchNotebookSection as ISettingsLayout['launchNotebookSection'];
+    },
+    label: trans.__('Show create new empty notebook section'),
```
Comment (krassowski) [inline src/commands.ts:111]:
```suggestion
    label: trans.__('Show "Create Notebook" section'),
```

This was an odd one:

image

File: src/launcher.tsx [hunk 4]
```diff
@@ -125,8 +169,36 @@ function LauncherBody(props: {
...
+            showWidgetType={true}
+            query={query}
+            settings={props.settings}
+            trans={trans}
+            onClick={item => item.execute()}
+            favouritesChanged={props.favouritesChanged}
+            lastUsedChanged={props.lastUsedChanged}
+          />
+        ) : (
+          'No starred items'
```
Comment (krassowski) [inline src/launcher.tsx:196]: Unrelated to this PR, but while we are at it:

```suggestion
          trans.__('No starred items')
```

**Comment by andrewfulton9** [comment]
@krassowski, thanks for the review! I updated all the settings text to properly reflect the actual section headers. Let me know if there is anything else I should do to close out this PR.

**Comment by krassowski** [comment]
JupyterLab 4.3.4 fixed the issue with "Incorrect locale information provided", I restarted the tests - let's see if everything passes.

**Comment by krassowski** [comment]
It looks like query strings in the visual regression test code need updating?

**Comment by andrewfulton9** [comment]
@krassowski, I'm not sure why the table-context-menu is failing. 
actual:
table-context-menu-actual
vs
expected:
table-context-menu-expected
  
Should I go ahead and update the images here anyway?

File: src/commands.ts [hunk 5]
```diff
@@ -76,19 +76,58 @@ export function addCommands(
...
       await settings.set('starredSection', !starredSection);
     }
   });
+  app.commands.addCommand(CommandIDs.showNotebookLauncher, {
+    isToggleable: true,
+    isToggled: () => {
+      return settings.composite
+        .launchNotebookSection as ISettingsLayout['launchNotebookSection'];
+    },
+    label: trans.__('Show "Launch New Notebook" section'),
```
Comment (krassowski) [inline src/commands.ts:111]: Nitpick: only this one had lowercase "section", aligning it now:

```suggestion
    label: trans.__('Show "Launch New Notebook" Section'),
```

**Comment by krassowski** [comment]
bot please update snapshots

**Review by krassowski — APPROVED** [review]
Thanks @andrewfulton9!

**Pull request merged**